torch/utils/cpp_extension.py

```diff
@@ -348,6 +348,18 @@ def finalize_options(self) -> None:
     def build_extensions(self) -> None:
         self._check_abi()
         for extension in self.extensions:
+            # Ensure at least an empty list of flags for 'cxx' and 'nvcc' when
+            # extra_compile_args is a dict. Otherwise, default torch flags do
+            # not get passed. Necessary when only one of 'cxx' and 'nvcc' is
+            # passed to extra_compile_args in CUDAExtension, i.e.
+            #   CUDAExtension(..., extra_compile_args={'cxx': [...]})
+            # or
+            #   CUDAExtension(..., extra_compile_args={'nvcc': [...]})
```

Comment:
Could you explain to me why this shouldn't be fixed downstream? It doesn't seem like having an entry with empty list or not having an entry should matter.

Reply:
I find it surprising that I have to pass `'cxx': []` when what I want is to set some flags for `nvcc` only.
I would find it more intuitive if the build would proceed with
`CUDAExtension(..., extra_compile_args={"nvcc": [...]})`
instead of requiring
`CUDAExtension(..., extra_compile_args={"cxx": [], "nvcc": [...]})`.

Reply:
@ezyang from checking this file. The different uses of this argument as a dictionary all assume that both keys are present.
So I think this is doing the right thing to initialize the entries that were not specified by the user.

Reply:
Landing this for now as I think there isn't a simple and reliable way to fix all the use sites.
@ezyang let me know if you think this is not the right fix and I will update.